Tuesdays/Week-Class/Inheratance/Assets/Scripts/TrapController.cs

```diff
@@ -3,18 +3,43 @@
 using UnityEngine;
 
 public class TrapController : MonoBehaviour{
-	[SerializeField] private Trap[] trapList;
     [SerializeField] private Transform[] spawnPoints;
-
+	[SerializeField] private Trap _trapPrefab;
+	private List<Trap.TrapDelegate> _trapBehaviours;
+	[SerializeField] TrapSprite[] _trapSprites;
+	[SerializeField] LayerMask[] _damageTarget;
+	
 	void Start(){
-		int rand;
-
-		foreach (Transform t in spawnPoints) {
-			rand = Random.Range(0, trapList.Length);
-			Trap randTrap = trapList[rand];
-			Instantiate(randTrap,t.position, t.rotation);
+		_trapBehaviours = new List<Trap.TrapDelegate>();
+		_trapBehaviours.Add(TrapBehaviours.DoDamage);
+		_trapBehaviours.Add(TrapBehaviours.TeleportBack);
+		_trapBehaviours.Add(TrapBehaviours.ChangeCharacter);
+		
+		Trap newTrap;
+		int randNum;
+		foreach(Transform t in spawnPoints) {
+			randNum = Random.Range(0, _trapBehaviours.Count);
+			newTrap = TrapFactory(_trapPrefab, _trapBehaviours[randNum], _trapSprites[randNum], _damageTarget[randNum]);
+			newTrap.transform.position = t.position;
 		}
         
     }
+
+	private void OnValidate() {
+		if(_trapSprites.Length != 3) {
```

Comment:
you don't really care about the number 3 right?
you care about checking that the sprites amount and the TrapBehaviours amount are the same